frontend/src/metabase/lib/i18n.js

```diff
@@ -58,15 +58,20 @@ function updateMomentStartOfWeek() {
 // if the start of week Setting is updated, update the moment start of week
 MetabaseSettings.on("start-of-week", updateMomentStartOfWeek);
 
-export function setLocalization(translationsObject) {
+function setLanguage(translationsObject) {
```

Comment:
The changes in this file are because I discovered that my previous approach of temporarily switching the global locales to the site locale and then switching it back doesn't work for momentjs, though it does for ttag. 

(For momentjs, this approach was causing _all_ dates on a dashboard to be formatted using the instance locale, even values in parameter widgets and elsewhere.)

Instead, I found [these docs](https://momentjscom.readthedocs.io/en/latest/moment/06-i18n/02-instance-locale/) which indicate an alternate approach for momentjs: we should just create a new `moment` object locally, set the locale on it to the site locale, and use it for formatting.

So I've changed the `withInstanceLocalization` helper to `withInstanceLanguage` and made it only affect the ttag locale. And I've udpated `substitute_tags` (in `parameters.cljc`) to take an optional locale argument to use for momentjs.

Reply:
> this approach was causing all dates on a dashboard to be formatted using the instance local

This seems scary, should we add a test for this tho? even though we already aware of this but I feel like it should be documented some where. a test might be a good idea.

Reply:
👍 I'll add a check to the cypress test